tests/sqla/test_view_serialization.py

```diff
@@ -0,0 +1,71 @@
+import uuid
+
+import pytest
+from sqlalchemy import Column, String
+from sqlalchemy.dialects.postgresql import UUID
+from sqlalchemy.orm import declarative_base
+from starlette.applications import Starlette
+from starlette.requests import Request
+from starlette_admin import RequestAction, StringField
+from starlette_admin.base import BaseAdmin
+from starlette_admin.contrib.sqla.view import ModelView
+
+Base = declarative_base()
+
+
+class User(Base):
+    __tablename__ = "user"
+
+    id: UUID = Column(
+        UUID(as_uuid=True), primary_key=True, default=uuid.uuid4, unique=True
+    )
+    name: str = Column(String())
+
+
+class UserView(ModelView):
+    fields = [
+        StringField("name"),
+    ]
+
+
+@pytest.fixture
+def user_view():
+    return UserView(User)
+
+
+@pytest.fixture
+def app(user_view: UserView):
+    app = Starlette()
+    app.state.ROUTE_NAME = "admin"
+
+    admin = BaseAdmin()
+    admin.add_view(user_view)
+    admin.mount_to(app)
+
+    return app
+
+
+@pytest.fixture
+def req(app: Starlette):
+    return Request(
+        {
+            "app": app,
+            "router": app.router,
+            "type": "http",
+            "headers": [],
+        }
+    )
+
+
+@pytest.mark.asyncio
+async def test_ensuring_pk(req: Request, user_view: UserView):
+    """
+    Ensures PK is present in the serialized data and properly serialized as a string.
+    """
+
+    user_id = uuid.uuid1()
+    user = User(id=user_id, name="Jack")
+
+    data = await user_view.serialize(user, req, RequestAction.LIST)
+
+    assert data["id"] == str(user_id)
```

Comment:
Starlette recommends [testclient](https://www.starlette.io/testclient/) to avoid creating the request object. I think It's better to simply call the listing API ,which will, in turn, invoke the serialize method. Here is an example -> https://github.com/jowilf/starlette-admin/blob/41a59ccdc61e0c5ce88f0a34436cbd0fb8d3309f/tests/sqla/test_sync_engine.py#L177

Reply:
This poses the question, why the serialization is dependent on the HTTP request object? Maybe consider decoupling them via repackaging parts of the request the serialization is dependent on into a separate DTO (the serialization context). This would give you exactly what I've done -- testing serialization thoroughly without executing fake HTTP requests.

I'll see how doing it through testclient looks for now.

Reply:
The request is used to generate some metadata as you can see here  https://github.com/jowilf/starlette-admin/blob/13e902fc06e1a68c0b2ba88b765aac569ba2aa4a/starlette_admin/views.py#L760

 > Maybe consider decoupling them via repackaging parts of the request the serialization is dependent on into a separate DTO (the serialization context).

This a good idea, can you open a discussion to present your approach?

Reply:
Hm, I'm seeing your commits on this. I was cleaning it up when seen them. )

Reply:
Hi, sorry I thought you were struggling with the test, and  I was trying to help. your implementation with the platform-independent UUID is better, the only thing missing is the relationship (for `foreign_model.get_ser...`) that you can copy and paste from my commit. Thanks

Reply:
Oh, ok. I was just busy yesterday. Will finish the PR changes today.